### ISSUE TITLE
Default parameters values are defined in multiple source files

### ISSUE THREAD

**Issue by ildoonet** [spragunr/deep_q_rl#9]
Hi

I'm very interested in this project and I'm also working on reinforcement learning on another project using this repository.

But since there are many parameters and they are defined in multiple source files (eg, default decay values are defined in function header and etc)

It is very hard to track the values.

What is your opinion to organize parameters so that we can easily track the values?

Thanks.

**Comment by AjayTalati** [comment]
Depends on what you're interested in? There are parameters which control the

i) Training the deep neural network - i.e. the solvers parameters, 
ii) The RL problem - discount factor, epsilon stuff, ..
iii) The memory problem - phi length
iv) The architecture of the network - i.e. the Q function appropriator - cuda_conv
v) You might want to redefine the loss function  
vi) Experiment setup parameters

I think you just need to focus on the things you find most interesting, adapt the code, and forget the rest?

**Comment by spragunr** [comment]
Hi @ildoonet .  I'm glad you are getting some use out of the package.  Here is the logic behind the current organization, such as it is: 

* `cnn_q_learner.py` - This is intended to encapsulate the deep-q-learning algorithm independent of any particular problem domain.  Default parameter values here will generally be overriden. 
* `rl_glue_ale_agent.py` - This file uses `cnn_q_learner.py` to create an agent for the ALE domain within the RLGlue framework.  If there are agent parameters that I want to be adjustable, I make them command line arguments for this executable, and I try to keep the defaults set to the best known values. 

How you manage parameter values, and which parameter values you are interested in, will largely depend on what you are doing with the package.   My priority has been on making sure that the parameters I'm most interested in can be adjusted without modifying the source code. 

I'm open to (easily-merged) pull requests or suggestions for improvement, but this isn't an issue I have a lot of time to work on.   You could also take a look at @alito's fork.  He has put some effort into general tidying-up, and he may have a better way of managing default parameter values.

**Comment by spragunr** [comment]
I'm going to close this, since there are no concrete plans to do anything about it.

**Comment by ildoonet** [comment]
Thanks!
